src/agent/coverage/src/block/windows.rs

```diff
@@ -139,38 +137,32 @@ impl BlockCoverageHandler {
         Ok(())
     }
 
-    fn try_on_poll(&mut self, dbg: &mut Debugger) -> Result<()> {
+    fn handle_on_poll(&mut self, dbg: &mut Debugger) {
         if !self.timed_out && self.started.elapsed() > self.max_duration {
             self.timed_out = true;
             dbg.quit_debugging();
         }
-
-        Ok(())
     }
 }
 
 impl DebugEventHandler for BlockCoverageHandler {
     fn on_create_process(&mut self, dbg: &mut Debugger, module: &Module) {
-        if self.try_on_create_process(dbg, module).is_err() {
+        if self.handle_on_create_process(dbg, module).is_err() {
             self.stop(dbg);
         }
     }
 
     fn on_load_dll(&mut self, dbg: &mut Debugger, module: &Module) {
-        if self.try_on_load_dll(dbg, module).is_err() {
-            self.stop(dbg);
-        }
+        self.handle_on_load_dll(dbg, module);
     }
 
     fn on_breakpoint(&mut self, dbg: &mut Debugger, bp: BreakpointId) {
-        if self.try_on_breakpoint(dbg, bp).is_err() {
+        if self.handle_on_breakpoint(dbg, bp).is_err() {
             self.stop(dbg);
         }
     }
 
     fn on_poll(&mut self, dbg: &mut Debugger) {
-        if self.try_on_poll(dbg).is_err() {
-            self.stop(dbg);
```

Comment:
This no longer calls self.stop on failure.  Is that intended?

Reply:
Yes, because that was unreachable. The original pattern was only mean to be a within-code reminder that if one of non-trait implementation methods were to fail, we should call `quit_debugging()`. The `on_poll()` method just exists to do exactly this _on timeout_ waiting for a debug event.